src/MongoDB/Manager.c

```diff
@@ -257,12 +256,12 @@ static PHP_METHOD(Manager, __construct)
 	/* Separate the options and driverOptions zvals, since we may end up
 	 * modifying them in php_phongo_manager_prep_uri_options() and
 	 * php_phongo_manager_merge_context_options() below, respectively. */
-	zend_replace_error_handling(EH_THROW, phongo_exception_from_phongo_domain(PHONGO_ERROR_INVALID_ARGUMENT), &error_handling);
-	if (zend_parse_parameters(ZEND_NUM_ARGS(), "|s!a/!a/!", &uri_string, &uri_string_len, &options, &driverOptions) == FAILURE) {
-		zend_restore_error_handling(&error_handling);
-		return;
-	}
-	zend_restore_error_handling(&error_handling);
+	PHONGO_PARSE_PARAMETERS_START(0, 3)
+	Z_PARAM_OPTIONAL
+	Z_PARAM_STRING_OR_NULL(uri_string, uri_string_len)
+	Z_PARAM_ARRAY_EX(options, 1, 1)
+	Z_PARAM_ARRAY_EX(driverOptions, 1, 1)
```

Comment:
Noted that `Z_PARAM_ARRAY_EX`'s second and third arguments correspond to `check_null` and `separate`, respectively (see: [lxr](https://heap.space/xref/PHP-8.1/Zend/zend_API.h?r=76fcd70c#1496)).